dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/preheat/DefaultPreheatService.java

```diff
@@ -432,6 +442,65 @@ public void validate( PreheatParams params )
         }
     }
 
+    /**
+     * Extracts and returns all uids from all dashboard items found in the given
+     * "objects".
+     *
+     * @param objects the map that contains lists of {@link IdentifiableObject}
+     * @return the set of dashboard items uid
+     */
+    private Set<String> collectDashboardItemsUid(
+        Map<Class<? extends IdentifiableObject>, List<IdentifiableObject>> objects )
+    {
+        if ( MapUtils.isEmpty( objects ) )
+        {
+            return emptySet();
+        }
+
+        Set<String> itemsUid = new HashSet<>();
+
+        for ( Map.Entry<Class<? extends IdentifiableObject>, List<IdentifiableObject>> entry : objects.entrySet() )
```

Comment:
can replace this for loop with `objects.get( Dashboard.class ) `

Reply:
Done